### PR TITLE
Switch to O prevew2 sdk with changes in media

### PR DESCRIPTION
**Pull request by dakegu**
1. Switch to use PlaybackTransportControlGlue + VideoFragment

2. added seek demo for video.

3. Switch music to use PlaybackFragment. Add next() and previous()
in music example, remove FF/RW since the five seconds jump model
isn't matching PlaybackControlGlue.

A lot of code clean up.

### CODE REVIEW

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java [hunk 1]
```diff
@@ -150,63 +105,56 @@ public void setupControlsRowPresenter(PlaybackControlsRowPresenter presenter) {
...
-        // If either 'Shuffle' or 'Repeat' has been clicked we need to make sure the acitons index
-        // is incremented and the UI updated such that we can display the new state.
-        super.onActionClicked(action);
-        if (action instanceof PlaybackControlsRow.ShuffleAction) {
-            mShuffleAction.nextIndex();
-        } else if (action instanceof PlaybackControlsRow.RepeatAction) {
-            mRepeatAction.nextIndex();
-        } else if (action instanceof PlaybackControlsRow.ThumbsUpAction) {
+        if (action instanceof PlaybackControlsRow.ShuffleAction
+                ||action instanceof PlaybackControlsRow.RepeatAction) {
```
Comment (WeiHsinChen) [inline app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java]: space after ||

Comment (WeiHsinChen) [inline app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java]: space after ||

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/VideoExampleActivity.java [hunk 2]
```diff
@@ -35,12 +35,9 @@ public void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_video_example);
 
-        FragmentTransaction ft1 = getFragmentManager().beginTransaction();
-        ft1.replace(R.id.videoFragment, new VideoSurfaceFragment(), VideoSurfaceFragment.TAG);
-        ft1.commit();
-
         FragmentTransaction ft2 = getFragmentManager().beginTransaction();
```
Comment (WeiHsinChen) [inline app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/VideoExampleActivity.java]: ft would be better

File: app/src/main/res/drawable/ic_favorite_filled_24dp.xml [hunk 3]
```diff
@@ -5,5 +5,5 @@
         android:viewportHeight="24.0">
     <path
         android:pathData="M12,21.35l-1.45,-1.32C5.4,15.36 2,12.28 2,8.5 2,5.42 4.42,3 7.5,3c1.74,0 3.41,0.81 4.5,2.09C13.09,3.81 14.76,3 16.5,3 19.58,3 22,5.42 22,8.5c0,3.78 -3.4,6.86 -8.55,11.54L12,21.35z"
-        android:fillColor="@color/song_row_favorite_color"/>
+        android:fillColor="#FF6E40"/>
```
Comment (WeiHsinChen) [inline app/src/main/res/drawable/ic_favorite_filled_24dp.xml:8]: It seems that color/song_row_favorite_color is still "#FF6E40" in this CL. Am I wrong?

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MusicMediaPlayerGlue.java [hunk 4]
```diff
@@ -394,44 +195,69 @@ private void queueNextRefresh() {
         return (mPlaybackService != null) ? mPlaybackService.getCurrentPosition() : 0;
     }
 
-    @Override protected void startPlayback(int speed) throws IllegalStateException {
+    @Override public void play(int speed) {
         prepareAndPlay(mMediaMetaData);
     }
 
-    @Override protected void pausePlayback() {
+    @Override public void pause() {
```
Comment (WeiHsinChen) [inline app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MusicMediaPlayerGlue.java:202]: Is there any reason to put Override with the method name in the same line?

File: app/src/main/res/drawable/ic_favorite_filled_24dp.xml (hunk 3 above)
Comment (dakegu) [reply app/src/main/res/drawable/ic_favorite_filled_24dp.xml:8]: it wont compile with new sdk

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MusicMediaPlayerGlue.java (hunk 4 above)
Comment (dakegu) [reply app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MusicMediaPlayerGlue.java:202]: All current methods are doing this way.  Might clean up later

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java (hunk 1 above)
Comment (dakegu) [reply app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/MediaPlayerGlue.java]: done

File: app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/VideoExampleActivity.java (hunk 2 above)
Comment (dakegu) [reply app/src/main/java/android/support/v17/leanback/supportleanbackshowcase/app/media/VideoExampleActivity.java]: done

File: app/build.gradle [hunk 5]
```diff
@@ -1,13 +1,13 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 25
-    buildToolsVersion "25"
+    compileSdkVersion 'android-O'
+    buildToolsVersion '26.0.0-rc2'
```
Comment (WeiHsinChen) [inline app/build.gradle:5]: Is '26.0.0-rc' already available?

**Review by WeiHsinChen — APPROVED** [review]
LGTM +1. Lets ship it :shipit:!

        /\/\

                        /\/\
         O  O  O
                 O
                __|__
                || ||_____
                || ||    |
     --------------------------
      \   O   O   O   O      /
 ~ ~ ~ ~ ~ ~ ~ ~  ~ ~ ~ ~ ~ ~ ~ ~
  ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~

**Pull request merged**